server/assetstore.py

```diff
@@ -199,6 +199,8 @@ def downloadFile(self, file, offset=0, headers=True, endByte=None,
                     'The extraParameters field must either be a dictionary, a '
                     'JSON-encoded dictionary, or a url query-encoded string.')
             params.update(extraParameters)
+            if params.get('limit', 'notpresent') is None:
```

Comment:
Yes.  Not passing a value is distinct from passing None (or javascript null).  If you don't pass a value, you get the default value (which, if not specified elsewhere is 50).